### PR TITLE
 Feature: Implement Nameable Class and Decorators

### PR DESCRIPTION
**Pull request by stino-x**
Hello,

I've implemented a new feature that enhances the Person class by introducing a Nameable module and decorator classes. Here's a summary of the changes:

Created the Nameable module:

The Nameable class  defines the correct_name method, which should be implemented by classes that include it.
Modified the Person class:

The Person class now includes the Nameable class
The correct_name method in Person simply returns the person's name.
Created the Decorator base class:

The Decorator class includes the Nameable class
It takes a Nameable object as a parameter during initialization and delegates the correct_name method to this object.
Created two specific decorators, CapitalizeDecorator and TrimmerDecorator:

CapitalizeDecorator capitalizes the name provided by the decorated object.
TrimmerDecorator ensures the name has a maximum of 10 characters and trims it if it exceeds this limit.
The decorator pattern allows you to add new behaviors to the Person class dynamically while preserving the core functionality. This design promotes flexibility and modularity in the code.

Feel free to review the changes, and I welcome any feedback or suggestions for improvement. Thank you!

Best regards,
[IHEAGWARA AUGUSTINE]

### CODE REVIEW

File: README.md
Comment (skyv26) [inline README.md:1]: - [ ] Please always ensure you have no broken links in the `README` file, as I tried to look into your `LICENSE` file but got a 404 error. I noticed that you forgot to add the file extension. Your file name is `LICENSE.md` but you are passing `LICENSE` only in the path. Kindly fix this issue, and cross-check that everything is working fine.

**Review by skyv26 — CHANGES_REQUESTED** [review]
## Hi @stino-x, 

Good job so far 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## STATUS: REQUIRE CHANGES 🔴 



## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implemented the features ✅ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

#### Cheers, and Happy coding!👏👏👏

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md
Comment (stino-x) [reply README.md:1]: i click on the link in line 145 and it went to the license i dont understand whats wrong here

Comment (stino-x) [reply README.md:1]: @skyv26

**Review by danmainah — APPROVED** [review]
Hi @stino-x , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️ Implemented the change requested and fixed the broken MIT license link.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by stino-x** [comment]
Thank you for your review, I would like to ask you for the code you used to test my work and if you used another file to do that 🙏 @danmainah

**Comment by danmainah** [comment]
> Thank you for your review, I would like to ask you for the code you used to test my work and if you used another file to do that 🙏 @danmainah

I used your person file then used the code below
require_relative 'capitalize_decorator'
require_relative 'trimmer_decorator' 

 person = Person.new(age: 22, name: 'maximilianus')
p person.correct_name
capitalized_person = CapitalizeDecorator.new(person)
p capitalized_person.correct_name
capitalized_trimmed_person = TrimmerDecorator.new(capitalized_person)
p capitalized_trimmed_person.correct_name

**Comment by stino-x** [comment]
hi, @danmainah thank you for your response i would like to ask you why you did it in the person file? could that est be mad e anywhere else ? thank you for your patience

**Comment by stino-x** [comment]
hi, @danmainah thank you for your response i would like to ask you why you did it in the person file ? could that est be mad e anywhere else ? thank you for your patience

**Comment by danmainah** [comment]
Just a preference you can add it anywhere

**Pull request merged**